File: src/main/java/raccoonlang/ArgumentsParser.java
package raccoonlang;

public class ArgumentsParser {

    public static boolean parseArguments(String[] args) {
        if (args.length < 1) {
            return false;
        }

        switch (args[0]) {
            case "compile"-> {
                System.out.println("Compile Time!");
            }
            case "help" -> {
                return false;
            }
            default -> {
                System.err.println("Option not recognized!");
                return false;
            }
        }

        return true;
    }

}

File: Args/ArgumentsParser.cs
namespace Raccoonlang.Args;

using Tokenizing;
using System.IO;

public class ArgumentsParser 
{
    public static bool ParseArguments(string[] args)
    {
        if (args.Length < 2) {
            return false;
        }

        switch(args[1]) {
            case "compile": 
                Tokenizer tokenizer = new Tokenizer();
                if (args.Length == 3) {
                    try {
                        string path = args[2];
                        tokenizer.Tokenize(path, File.ReadAllText(path));
                    } catch (Exception e)
                    {
                        throw;
                    }
                }
                break;
            case "help":
                return true;
            default:
                Console.WriteLine("Don't recognize this option!");
                return false;
        }

        return true;
    }
}